modules/log/log.go

```diff
@@ -264,11 +264,13 @@ func (l *Logger) StartLogger() {
 	for {
 		select {
 		case bm := <-l.msg:
+			l.lock.Lock()
 			for _, l := range l.outputs {
 				if err := l.WriteMsg(bm.msg, bm.skip, bm.level); err != nil {
 					fmt.Println("ERROR, unable to WriteMsg:", err)
 				}
 			}
+			l.lock.Unlock()
```

Comment:
I expected defer is function scoped. Correct me if I am wrong.
I am not sure about deadlocks detection of -race. I suppose it doesn't help.